pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -55,9 +70,118 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 			}
 		}
 	}
+	return nil
+}
 
-	instance.Status.StorageClassesCreated = true
-
+// ensureExternalCluster ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalCluster(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalClusterDetailsSecret,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
```

Comment:
I think we should check for `if errors.IsNotFound(err) {` and potentially log something useful to show up in the op logs.